nautobot/utilities/filters.py

```diff
@@ -454,6 +460,29 @@ class BaseFilterSet(django_filters.FilterSet):
         }
     )
 
+    class Meta:
+        """Base class that all BaseFilterSet.Meta subclasses must inherit from."""
+
+        class StrictFilterForm(forms.Form):
+            """Form class that treats any unknown form data entries as a potential validation error."""
+
+            def is_valid(self):
+                result = super().is_valid()
+                for extra_key in set(self.data.keys()).difference(self.cleaned_data.keys()):
+                    # If a given field was invalid, it will be omitted from cleaned_data; don't report extra errors
+                    if extra_key not in self.errors:
+                        if settings.STRICT_FILTERING:
+                            # Sure would be nice if we could instead do:
+                            # self.add_error(extra_key, "Unknown filter field") but extra_key isn't a field, so...
+                            self.add_error(None, f'Unknown filter field "{extra_key}"')
+                            result = False
+                        else:
+                            logger.warning('%s: Unknown filter field "%s"', self.__class__.__name__, extra_key)
+
+                return result
+
+        form = StrictFilterForm
```

Comment:
I see where this is going, however I am concerned about this adding the requirement to subclass `BaseFilterSet.Meta` on every subclass. I am wondering if there might be a better way to get the same behavior here such as overloading [`BaseFilterSet.get_form_class()`](https://github.com/carltongibson/django-filter/blob/78210722d920f803a8142e48969bc37f0f8324ed/django_filters/filterset.py#L247) or maybe subclassing the [filterset metaclass](https://github.com/carltongibson/django-filter/blob/78210722d920f803a8142e48969bc37f0f8324ed/django_filters/filterset.py#L65) (not to be confused with inner `Meta` class).